src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/CryptoConfig.cs

```diff
@@ -0,0 +1,443 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+
+namespace System.Security.Cryptography
+{
+    public class CryptoConfig
+    {
+        private const string AssemblyName_Cng = "System.Security.Cryptography.Cng";
+        private const string AssemblyName_Csp = "System.Security.Cryptography.Csp";
+        private const string AssemblyName_Pkcs = "System.Security.Cryptography.Pkcs";
+        private const string AssemblyName_X509Certificates = "System.Security.Cryptography.X509Certificates";
+
+        private const BindingFlags ConstructorDefault = BindingFlags.Instance | BindingFlags.Public | BindingFlags.CreateInstance;
+
+        private const string OID_RSA_SMIMEalgCMS3DESwrap = "1.2.840.113549.1.9.16.3.6";
+        private const string OID_RSA_MD5 = "1.2.840.113549.2.5";
+        private const string OID_RSA_RC2CBC = "1.2.840.113549.3.2";
+        private const string OID_RSA_DES_EDE3_CBC = "1.2.840.113549.3.7";
+        private const string OID_OIWSEC_desCBC = "1.3.14.3.2.7";
+        private const string OID_OIWSEC_SHA1 = "1.3.14.3.2.26";
+        private const string OID_OIWSEC_SHA256 = "2.16.840.1.101.3.4.2.1";
+        private const string OID_OIWSEC_SHA384 = "2.16.840.1.101.3.4.2.2";
+        private const string OID_OIWSEC_SHA512 = "2.16.840.1.101.3.4.2.3";
+        private const string OID_OIWSEC_RIPEMD160 = "1.3.36.3.2.1";
+
+        private static volatile Dictionary<string, string> defaultOidHT = null;
+        private static volatile Dictionary<string, object> defaultNameHT = null;
+
+        // CoreFx does not support AllowOnlyFipsAlgorithms
+        public static bool AllowOnlyFipsAlgorithms => false;
+
+        private static Dictionary<string, string> DefaultOidHT
+        {
+            get
+            {
+                if (defaultOidHT != null)
+                {
+                    return defaultOidHT;
+                }
+
+                Dictionary<string, string> ht = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+                ht.Add("SHA", OID_OIWSEC_SHA1);
+                ht.Add("SHA1", OID_OIWSEC_SHA1);
+                ht.Add("System.Security.Cryptography.SHA1", OID_OIWSEC_SHA1);
+                ht.Add("System.Security.Cryptography.SHA1CryptoServiceProvider", OID_OIWSEC_SHA1);
+                ht.Add("System.Security.Cryptography.SHA1Cng", OID_OIWSEC_SHA1);
+                ht.Add("System.Security.Cryptography.SHA1Managed", OID_OIWSEC_SHA1);
```

Comment:
Who would do that mapping to SHA1Cng? The Oid dictionary only used in MapNameToOid, not CreateFromName (unless I'm missing something)

Note however that there is a pending discussion\resolution on how to handle Unix\Mac w.r.t. return Csp types instead of compatible type (like *Managed type), although we want Csp types to eventually be supported by Unix\Mac so likely we won't change anything.